
  >>> from Products.Five.testbrowser import Browser
  >>> browser = Browser()
  >>> browser.handleErrors = False
  >>> self.portal.error_log._ignored_exceptions = ()


Lets try to invalidate cache on a folder

  >>> browser.mech_browser.set_handle_redirect(False)
  >>> from urllib2 import HTTPError
  >>> try:
  ...   browser.open(self.portal.SITE.absolute_url() + '/@@invalidate_cache')
  ... except HTTPError, exc:
  ...   print str(exc)
  HTTP Error 401: Unauthorized

 
This didn't work since we went directly to the url but it's only allowed 
when you click on the link from the same context. The HTTP_REFERER has to be
absolute url of context. This is to avoid links from search results if robots
are clever enough to get the url. The invalidate_cache should be redirected 
to context.


If we press the link on the current page we should invalidate the cache
and get redirected back to the same page with a status message.

  >>> browser.open(self.portal.SITE.absolute_url())
  >>> invalidateLink = browser.getLink(text='Refresh this page')

  >>> from urllib2 import HTTPError
  >>> try:
  ...   invalidateLink.click()
  ... except HTTPError, exc:
  ...   print str(exc)
  ...   print exc.headers.getheader('Location')
  HTTP Error 302: Moved Temporarily
  http://nohost/plone/SITE?portal_status_message=Request for cache invalidation sent

There is also a test that the IPnumber is EEA internal or 127.0.0.1 for 
development/testing. This should prevent robots and users from outside of EEA to
trigger cache invalidation.


  >>> id = self.portal.invokeFactory('Document', id='doc1', title='doctitle')
  >>> doc = self.portal[id]
  >>> doc.reindexObject()
  >>> self.portal.portal_workflow.doActionFor(doc, 'publish')
  >>> browser.open(doc.absolute_url())
  >>> invalidateLink = browser.getLink(text='Refresh this page')
  >>> from urllib2 import HTTPError
  >>> try:
  ...   invalidateLink.click()
  ... except HTTPError, exc:
  ...   print str(exc)
  ...   print exc.headers.getheader('Location')
  HTTP Error 302: Moved Temporarily
  http://nohost/plone/doc1?portal_status_message=Request for cache invalidation sent
  
